system/ee/ExpressionEngine/Addons/structure/views/index.php

```diff
@@ -145,10 +145,19 @@
                     }
                 }
 
-                if ($page['hidden'] == 'y') {
+                if (isset($settings['show_entry_status']) && $settings['show_entry_status'] == 'y') {
+                    $status = ee('Model')->get('Status')->filter('status', '==', $page['status'])->first();
+                    echo $status->renderTag();
+                }
```

Comment:
I think this part needs to be optimized. If you have 50 entries, that would mean 50 extra queries, most of which will be duplicates. It should be a lot better to eager load statuses before sending variables to view file (and then maybe run `array_map` on result while calling `renderTag` to get extra variable for status badge)
Also there can be cases where status has been removed from system, but entries still exist with that status - we need to make sure to not break things if that's the case.

Reply:
I can absolutely optimize this by getting all Statuses outside of the loop and then referencing them here, but how would you suggest handling cases where the status has been removed, but entries still exist with that status? How do other parts of the CMS handle this?

Reply:
Here's one example (not eager loading here though) :
```
!empty($entry->getStatus()) ? $entry->getStatus()->renderTag() : $entry->status
```

Reply:
Here's the default style. I think this is slightly cleaner for UI consistency:

<img width="246" alt="image" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/2423727/2dacbfcc-f8ee-4fe1-bae8-00f11b00bf70">